src/index.ts

```diff
@@ -3,9 +3,22 @@ export type GoResultSuccess<T> = { data: T; success: true };
 export type GoResultError<E extends Error = Error> = { error: E; success: false };
 export type GoResult<T, E extends Error = Error> = GoResultSuccess<T> | GoResultError<E>;
 
+export interface StaticDelayOptions {
+  type: 'static';
+  delayMs: number;
+}
+
+export interface RandomDelayOptions {
+  type: 'random';
+  minDelayMs: number;
+  maxDelayMs: number;
+}
+
 export interface GoAsyncOptions {
-  readonly retries?: number;
-  readonly timeoutMs?: number;
+  readonly retries?: number; // Number of retries to attempt if the go callback is unsuccessful.
```

Comment:
Do we want the `readonly` modifiers here?

Reply:
I don't think it hurts, but I can remove them if you want.

Reply:
I generally like immutable everything (I like strict functional programming a fair amount 😄). However, I think for utility libraries, these restrictions might get in the way for some users.

But regardless, maybe it should be consistent with `StaticDelayOptions` and `RandomDelayOptions`, which don't currently have the `readonly` modifiers.

Reply:
I am removing them.